provisioning/WFGP-292_doc_galleon-pack.adoc

```diff
@@ -0,0 +1,161 @@
+** Add a `doc` goal to the maven plugin to generate a companion doc archive next to the feature pack itself
+* Galleon - https://github.com/wildfly/galleon
+** Might be impacted if we are missing any metadata from the feature pack to generate its documentation
+* Wildscribe - https://github.com/wildfly/wildscribe
+** Reuse its site-generator to generate the Model reference of the feature packs
+* Any project publishing feature packs (wildfly, grpc-feature-pack, ai-feature-pack, etc.)
+** They will be updated to bump the version of the Galleon maven plugin
+** They will have to execute the `doc` goal, *in addition to the `build-feature-pack` goal*
+** *No other changes should be required*
+* WildFly - https://github.com/wildfly/
+** Its model references will be generated when its feature packs are built. The dependency to wildscribe from its docs module will be removed 
+** WildFly release process will have to be updated so that its model reference is published to https://docs.wildfly.org/ based on this new doc archive instead of the model reference generated by wildfly docs module.
+
+=== Relevant Installation Types
+
+This requirement does not impact installation types.
+
+== Requirements
+
+
+* When the `doc` goal of the `org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin` is invoked (i.e. when the feature pack is built), generate a `doc.zip` archive that contains:
+** human-readable HTML documention of the feature pack stored in a `doc` directory. This documentation includes:
+*** the Model reference of the resources provided by the feature pack (in a format similar to the existing https://docs.wildfly.org/36/wildscribe/[WildFly model reference])
+*** the log messages that can be displayed by the feature pack (in a format similar to the existing https://docs.wildfly.org/36/wildscribe/log-message-reference.html[WildFly log messages])
+** the model reference of the resources provided by the feature pack (stored in a DMR representation in a JSON file) - generated during the `build-feature-pack` goal and stored in a `META-INF/model.json` file
+** the metadata of the feature pack (stored in a JSON file) - generated during the `build-feature-pack` goal and stored in a `META-INF/metadata.json` file
+* This companion doc.zip is attached to the feature pack's Maven model with the coordinates:
+** `groupId` - same as the feature pack
+** `artifactId` - same as the feature pack
+** `version` - same as the feature pack
+** `classifier` - `doc`
+** `extension` - `zip`
+* When the feature pack is installed or published, this companion doc.zip is also installed and published
+* Other tools that wants to leverage this documentation artifacts must not expect they are always availavle alongside the feature pack. It will only exist when feature packs project are upgraded to the Galleon Maven plugin that provides this feature. 
+
+=== Non-Requirements
+
+* the Look and Feel of the generated documenation is not configurable (it is based on wildscribe L&F).
```

Comment:
This should specify that coverage of Galleon constructs besides the feature pack itself is not in scope.

* Config
* *Layer
* Feature
* Feature group
* Package

If any of those are in scope, the requirements section should describe the details.

Reply:
yes, I've updated the requirements section.